`Development`: Add documentation for standardized competencies (#8711)
ls1intum/Artemis: docs/user/adaptive-learning/adaptive-learning-instructor.rst

Manage Adaptive Learning (Instructor)
=======================================

.. contents:: Content of this document
    :local:
    :depth: 2

.. note::

    When introducing adaptive learning into a course for the first time, you can quickly create a competency model by:

>     - :ref:`Generating competencies using Iris<generate_competencies>`

Reviewer: ```suggestion
    - :ref:`Generating competencies using an integrated LLM<generate_competencies>`
```

    (Given that these features are enabled on your Artemis instance)

Competencies
------------
A competency is a overarching learning objective that ties together various lectures and exercises. It has a title, description and a `taxonomy <https://en.wikipedia.org/wiki/Bloom%27s_taxonomy>`_.
By completing lectures and exercises, students gain mastery of a competency (see also :ref:`Competencies for Students<competencies_student>`).

A prerequisite is a competency that students are expected to have already mastered before the course. Instructors select competencies from previous courses they taught as a prerequisite, or create new ones.

Manage Competencies
^^^^^^^^^^^^^^^^^^^^
Instructors can manage competencies and prerequisites of a course in the *Competency Management* view. Here they can:

* View all competencies and prerequisites of their course
* Create, edit or delete competencies and prerequisites
* Manage relations between competencies (which are necessary to build learning paths)
* :ref:`import_competencies` from other courses or the :ref:`standardized competency catalog<standardized_competency_catalog>`

> * :ref:`generate_competencies` using Iris

Reviewer: ```suggestion
* :ref:`generate_competencies` using LLMs
```

|instructor-import-all-competencies|

**2. Import from other Courses**

| When importing from other courses, instructors can view a list of competencies from all courses they have access to.
  They can sort the list based on all fields except for competency description.
  They can also filter the list through the *Competency Search* at the top of the page, which can be expanded for advanced search options.
| All selected competencies are displayed in the *Selected Competencies* table and can be de-selected again by clicking on the trash icon.
  Finally, instructors can select if they also want to import the relations between the imported competencies.

|instructor-import-competencies|

.. _instructor_import_standardized_competencies:

**3. Import Standardized Competencies**

| When importing standardized competencies, instructors can view the complete :ref:`standardized competency catalog<standardized_competency_catalog>` of their Artemis instance.
  They can filter the displayed competencies by title and knowledge area.
| Clicking on the name of a competency opens its details on the right side and clicking on the checkbox next to the name selects it for import.

>   Again, selected competencies are displayed in a table and can be de-selected again by clicking on the trash icon.

Reviewer 2: ```suggestion
  Again, selected competencies are displayed in a table and can be de-selected again by clicking on the bin icon.
```

|instructor-import-standardized-competencies|

.. _generate_competencies:

Generate Competencies
^^^^^^^^^^^^^^^^^^^^^

.. raw:: html

    <iframe src="https://live.rbg.tum.de/w/artemisintro/46941?video_only=1&t=0" allowfullscreen="1" frameborder="0" width="600" height="350">
        Watch this video on TUM-Live.
    </iframe>

> | Instructors can use Iris, the AI subsystem for Artemis, to generate competencies for their course.

Reviewer: same here

> | Instructors can use Iris, the AI subsystem for Artemis, to generate competencies for their course.

Author: ```suggestion
| Instructors can use the integrated LLM subsystem of Artemis, to generate competencies for their course.
```

> | Based on a course description, Iris generates around 10 recommendations for possible competencies (the amount may vary depending on the description provided).

Reviewer: same here

> | Based on a course description, Iris generates around 10 recommendations for possible competencies (the amount may vary depending on the description provided).

Author: ```suggestion
| Based on a course description, the LLM generates around 10 recommendations for possible competencies (the amount may vary depending on the description provided).
```

Manage Adaptive Learning (Instructor)
=======================================

.. contents:: Content of this document
    :local:
    :depth: 2

.. note::

    When introducing adaptive learning into a course for the first time, you can quickly create a competency model by:

    - :ref:`Generating competencies using an integrated LLM<generate_competencies>`
    - :ref:`Importing standardized competencies into your course<instructor_import_standardized_competencies>`

    (Given that these features are enabled on your Artemis instance)

Competencies
------------

> A competency is a overarching learning objective that ties together various lectures and exercises. It has a title, description and a `taxonomy <https://en.wikipedia.org/wiki/Bloom%27s_taxonomy>`_.

Reviewer 3: ```suggestion
A competency is an overarching learning objective that ties together various lectures and exercises. It has a title, description and a `taxonomy <https://en.wikipedia.org/wiki/Bloom%27s_taxonomy>`_.
```
